### ISSUE TITLE
DL3045: False positive on windows file path

### ISSUE THREAD

**Issue by dan-neumegen-xero** [hadolint/hadolint#580]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Using a hadolint docker image (hadolint/hadolint:v1.23.0-8-gb01c5a9-alpine)

When linting the following 
`COPY . C:/test`

No warnings/errors are given

### Actual behavior

Using a hadolint docker image (hadolint/hadolint:v1.23.0-8-gb01c5a9-alpine)

When linting the following 
`COPY . C:/test`

The following is 
./Dockerfile:9 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.

### Steps to reproduce the behavior

Hadolint doesn't seem to process windows paths using forward slashes (which though technically incorrect still work)
To reproduce run hadolint over this docker file
``` Dockerfile
FROM mcr.microsoft.com/windows/nanoserver:20H2

COPY . C:/test
COPY . C:\test
```
And see that it picks up one of the first line but not the second.

This is a vary minor issue and not a blocker (just change your slashes of course) but letting you know anyway

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.23.0-8-gb01c5a9
```

**Additional environment details (OS, stack version, etc.)**
Windows 10 PC, running linux hadolint container (hadolint/hadolint:v1.23.0-8-gb01c5a9-alpine)
